tests_e2e/tests/scripts/agent_firewall-verify_all_firewall_rules.py

```diff
@@ -0,0 +1,427 @@
+        os.seteuid(uid)
+    except Exception as e:
+        raise Exception("Error -- failed to switch user to {0} : Failed with exception {1}".format(user, e))
+
+
+def get_root_accept_rule(command: str) -> List[str]:
+    return ['sudo', 'iptables', '-t', 'security', command, 'OUTPUT', '-d', get_wireserver_ip(), '-p', 'tcp', '-m',
+            'owner',
+            '--uid-owner',
+            '0', '-j', 'ACCEPT', '-w']
+
+
+def get_non_root_accept_rule(command: str) -> List[str]:
+    return ['sudo', 'iptables', '-t', 'security', command, 'OUTPUT', '-d', get_wireserver_ip(), '-p', 'tcp',
+            '--destination-port', '53', '-j',
+            'ACCEPT', '-w']
+
+
+def get_non_root_drop_rule(command: str) -> List[str]:
+    return ['sudo', 'iptables', '-t', 'security', command, 'OUTPUT', '-d', get_wireserver_ip(), '-p', 'tcp', '-m',
+            'conntrack', '--ctstate',
+            'INVALID,NEW', '-j', 'DROP', '-w']
+
+
+def execute_cmd(cmd: List[str]):
+    """
+    Note: The shellutil.run_command don't return the exit code of the command. So using subprocess.Popen
+    """
+    proc = subprocess.Popen(cmd,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            shell=False)
+    stdout, stderr = proc.communicate()
+    return proc.returncode, stdout, stderr
+
+
+def check_if_iptable_rule_is_available(full_command: List[str]) -> bool:
+    """
+    This function is used to check if given rule is present in iptable rule set
+    "-C" return exit code 0 if the rule is available.
+    """
+    exit_code, _, _ = execute_cmd(full_command)
+    if exit_code == 0:
+        return True
+    return False
+
+
+def print_current_iptable_rules():
+    """
+    This function prints the current iptable rules
+    """
+    try:
+        cmd = ['sudo', 'iptables', '-L', 'OUTPUT', '-t', 'security', '-nxv']
+        exit_code, stdout, stderr = execute_cmd(cmd)
+        if exit_code != 0:
+            log.warning("Warning -- Failed to fetch the ip table rules with error code: %s and error: %s", exit_code, stderr)
+        else:
+            for line in stdout.splitlines():
+                log.info(str(line))
+    except Exception as error:
+        raise Exception("Error -- Failed to fetch the ip table rule set {0}".format(error))
+
+
+def get_all_iptable_rule_commands(command: str) -> Tuple[List[str], List[str], List[str]]:
+    return get_root_accept_rule(command), get_non_root_accept_rule(command), get_non_root_drop_rule(command)
+
+
+def verify_all_rules_exist(max_retry: int = 5) -> None:
+    """
+    This function is used to verify all the iptable rules are present in the rule set
+    """
+    log.info("-----Verifying all ip table rules are present in rule set")
+    retry: int = 0
+    found: bool = False
+    while retry < max_retry and not found:
+        root_accept, non_root_accept, non_root_drop = get_all_iptable_rule_commands("-C")
```

Comment:
can you create an enum with -C, -D, etc and use it in these calls? will make the code easier to read